app/src/androidTest/java/net/skyscanner/backpack/starrating/BpkStarRatingTest.kt

```diff
@@ -0,0 +1,105 @@
+package net.skyscanner.backpack.starrating
+
+import android.graphics.Color
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import net.skyscanner.backpack.BpkSnapshotTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+
+class BpkStarRatingTest : BpkSnapshotTest() {
+
+  private val rating = BpkStarRating(testContext).apply { setBackgroundColor(Color.WHITE) }
+
+  @Before
+  fun setup() {
+    setDimensions(16, 80)
+  }
+
+  @Test
+  fun screenshotTestStarRating_Default() {
+    snap(rating)
+  }
+
+  @Test
+  fun screenshotTestStarRating_CustomMaxRatingIncreasing() {
+    setDimensions(16, 160)
+    rating.maxRating = 5
+    rating.maxRating = 10
+    snap(rating)
+  }
+
+  @Test
+  fun screenshotTestStarRating_CustomMaxRatingDecreasing() {
+    rating.maxRating = 5
+    rating.maxRating = 3
+    snap(rating)
+  }
+
+  @Test
+  fun screenshotTestStarRating_NegativeValue() {
+    rating.maxRating = 5
+    rating.rating = -0.5f
+    snap(rating)
+  }
+
+  @Test
+  fun screenshotTestStarRating_ZeroValue() {
+    rating.maxRating = 5
+    rating.rating = 0.0f
+    snap(rating)
+  }
+
+  @Test
+  fun screenshotTestStarRating_BeforeHalfStarValue() {
```

Comment:
Can we change the name of those tests? I'm not sure what Before/OverHalf means here. We can name it something like `RatingValueBetween0And0_5` for example

Reply:
Done